product_supplierinfo_for_customer_sale/models/product_customerinfo.py

```diff
@@ -0,0 +1,9 @@
+from odoo import api, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "product.customerinfo"
+
+    @api.onchange("product_tmpl_id")
+    def onchange_product_tmpl_id(self):
```

Comment:
This shouldn't be done this way, but with a computed writable.

Reply:
Hi, It serves to enhance the user experience, clearing the field and requiring them to select a new variant.